### ISSUE TITLE
no-unused-variable fails to recognize jsx (tsx) tags

### ISSUE THREAD

**Issue by ddaghan** [palantir/tslint#1034]
```ts
import foo = require('foo')

export function render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
	return <foo>
		hello world
	</foo>;
}
```
\>\> (no-unused-variable) file.tsx[1, 8]: unused variable: 'foo'

foo is used within jsx (tsx) but tslint does not recognize it.

**Comment by adidahiya** [comment]
@ddaghan which TSLint and TypeScript versions are you using?

**Comment by adidahiya** [comment]
Also, what kind of component is `foo`? Is it class-based or is it an SFC? Possibly related is https://github.com/Microsoft/TypeScript/issues/7109

**Comment by ddaghan** [comment]
typescript 1.7.5
tslint 3.2.1
grunt-tslint 3.0.1

correction for foo -> Foo

```
import Foo = require('Foo')

export function render(): ReactTypes.ReactElement<ReactTypes.HTMLAttributes> {
    return <Foo>
        hello world
    </Foo>;
}
```
It seems it is related to that

**Comment by JKillian** [comment]
@ddaghan Just tested your example code with `typescirpt@1.9.0-dev.20160322` (the latest version) the issue is resolved now. (I also verified that I could reproduce your issue with an older version of TypeScript).
